modules/javafx.controls/src/main/java/javafx/scene/chart/Chart.java

```diff
@@ -511,6 +511,71 @@ public StyleableProperty<Boolean> getStyleableProperty(Chart node) {
         return getClassCssMetaData();
     }
 
-}
+    private void handleAccessibilityActive(boolean on) {
+        titleLabel.setFocusTraversable(on);
+        updateSymbolFocusable(on);
+    }
 
+    /**
+     * Invoked in the FX application thread when accessibility active platform property changes.
+     * The child classes should override this method to set focus traversable flag on every symbol, as well as
+     * other elements that need to be focusable.
+     * @param on whether the accessibility is active
+     */
+    // package protected: custom charts must handle accessbility on their own
+    void updateSymbolFocusable(boolean on) {
+    }
 
+    /**
+     * When called from JavaFX application thread, returns the value of the property.
+     * When called from any other thread, returns false and sets up the machinery to
+     * invoke {@code updateSymbolFocusable()} when needed.
+     * The chart implementations should use this method to set focus travesable flags on the nodes
+     * which needs to be focus traversable when accessibility is on.
+     * @return
+     */
+    // package protected: custom charts must handle accessbility on their own
+    final boolean isAccessibilityActive() {
+        if (Platform.isFxApplicationThread()) {
+            if (accessibilityActive instanceof SimpleBooleanProperty p) {
+                return p.get();
+            } else {
+                boolean aa = Platform.accessibilityActiveProperty().get();
+                SimpleBooleanProperty active = new SimpleBooleanProperty(aa);
+                accessibilityActive = active;
+                active.bind(Platform.accessibilityActiveProperty());
+                active.addListener((src, prev, on) -> {
+                    handleAccessibilityActive(on);
+                });
+                return active.get();
```

Comment:
I'm confused as to what is happening here.  A boolean property `active` is created, which is bound to `Platform.accessibilityActiveProperty` (this creates a listener on `accessibilityActiveProperty`). We add then another listener on `active`.  

What confuses me is what this is supposed to achieve, and also why we're initialising the property, but then bind it anyway... `active = new SimpleBooleanProperty(aa)` and later `active.bind` means that the variable `aa` is completely unnecessary as the `bind` will do this `get` for you.

Reply:
aa is unnecessary, you are right, bind() will set the value.

the property is created as a way to signal any subsequent calls that no more work is needed.